### PR TITLE
New ``mode`` argument replacing ``direction``, Calculate scaled dimensions

### PR DESCRIPTION
**Pull request by thet**
Continuation of: #37, #36, #29

### CODE REVIEW

**Comment by thet** [comment]
@jenkins-plone-org please run jobs

File: news/xx.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix documentation of scaling modes to match it's behavior.
```
Comment (thet) [inline news/xx.bugfix]: After the test run this filename should be changed to ``39``

**Comment by thet** [comment]
@jenkins-plone-org please run jobs

**Comment by thet** [comment]
@mauritsvanrees tnx for the review. The tests were build following the logic of the documentation before I changed it and it failed: https://github.com/plone/plone.scale/pull/39/commits/d4acc6e6adc90ad495cb38539a5bc016b2905d3c
Simply reversing it like you suggested made them pass: https://github.com/plone/plone.scale/pull/39/commits/76f9c08fde62a04331fe96aef89e91e1125e80bd

I'm merging. Any improvements should be done in a separate PR.

**Comment by thet** [comment]
Hurray! A 2,5 years journey has come to an end!

**Comment by mauritsvanrees** [comment]
Thanks a lot @thet and @fschulze !

**Pull request merged**